Index: R4/nightly
===================================================================
--- R4/nightly	(revision 4297)
+++ R4/nightly	(working copy)
@@ -13,6 +13,8 @@
 # NOTE: packaging step re-signs all binaries and adds entitlements,
 #       so packaged version is not identical to the original build!
 
+set -e
+
 : ${BASE=/Volumes/Builds/R4}
 : ${RDIRS=`cat $BASE/builds`}
 
Index: R4/packaging/build.sh
===================================================================
--- R4/packaging/build.sh	(revision 4297)
+++ R4/packaging/build.sh	(working copy)
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 : ${BASE=/Volumes/Builds/R4}
 
 PKGROOT="$BASE/packaging"
@@ -43,27 +45,27 @@
 
 echo " - Removing previous builds ..."
 rm -rf "$DST/R-fw" "$DST/R-app"
-mkdir "$DST/R-fw" "$DST/R-app"
+mkdir -p "$DST/R-fw" "$DST/R-app"
 rm -f "$PKGROOT/R-app.pkg" "$PKGROOT/R-fw.pkg" "$PKGROOT/R.pkg"
 
 echo " - Extracting R.app ..."
 tar fxz "$Rapp" -C "$DST/R-app"
 
-. "$BASE/unlock-sign"
+#. "$BASE/unlock-sign"
+#
+#echo " - Signing R.app ..."
+#"$PKGROOT/R-sign-contents" "$DST/R-app" --force
+### after signing inside contents we also have to re-sign the entire app bundle
+#codesign --force -o runtime --timestamp --entitlements "$BASE/R.entitlements" -s "Developer ID Application" "$DST/R-app/R.app"
+#
+#echo " - Checking R.app signature ..."
+#if codesign -v "$DST/R-app/R.app"; then
+#    echo "   OK"
+#else
+#    echo "** ERROR: $DST/R-app/R.app does not pass codesign validation" >&1
+#    exit 1
+#fi
 
-echo " - Signing R.app ..."
-"$PKGROOT/R-sign-contents" "$DST/R-app" --force
-## after signing inside contents we also have to re-sign the entire app bundle
-codesign --force -o runtime --timestamp --entitlements "$BASE/R.entitlements" -s "Developer ID Application" "$DST/R-app/R.app"
-
-echo " - Checking R.app signature ..."
-if codesign -v "$DST/R-app/R.app"; then
-    echo "   OK"
-else
-    echo "** ERROR: $DST/R-app/R.app does not pass codesign validation" >&1
-    exit 1
-fi
-
 echo " - Extracting R framework ..."
 tar fxz "$FWTAR" -C "$DST/R-fw"
 mv "$DST/R-fw/Library/Frameworks/R.framework" "$DST/R-fw/"
@@ -70,7 +72,7 @@
 rm -rf "$DST/R-fw/Library"
 
 echo " - Signing R framework ..."
-"$PKGROOT/R-sign-contents" "$DST/R-fw"
+# "$PKGROOT/R-sign-contents" "$DST/R-fw"
 
 GUIVER=`sed -n 's:.*<string>R .*GUI ::p' "$DST/R-app/R.app/Contents/Info.plist" | sed 's:[- ].*::' | head -n1`
 if [ -z "$GUIVER" ]; then
@@ -82,7 +84,8 @@
 
 echo " - Building R.app package ..."
 echo pkgbuild --sign 'Developer ID Installer' --identifier org.R-project.R.GUI.pkg --install-location /Applications --version "$GUIVER" --timestamp --root "$DST/R-app" --component-plist "$PKGROOT/R-app.plist" "$PKGROOT/R-app.pkg"
-pkgbuild --sign 'Developer ID Installer' --identifier org.R-project.R.GUI.pkg --install-location /Applications --version "$GUIVER" --timestamp --root "$DST/R-app" --component-plist "$PKGROOT/R-app.plist" "$PKGROOT/R-app.pkg"
+echo pkgbuild --identifier org.R-project.R.GUI.pkg --install-location /Applications --version "$GUIVER" --timestamp --root "$DST/R-app" --component-plist "$PKGROOT/R-app.plist" "$PKGROOT/R-app.pkg"
+pkgbuild --identifier org.R-project.R.GUI.pkg --install-location /Applications --version "$GUIVER" --timestamp --root "$DST/R-app" --component-plist "$PKGROOT/R-app.plist" "$PKGROOT/R-app.pkg"
 
 RVER0=`sed -n 's:.*R_MAJOR::p' < "$DST/R-fw/R.framework/Headers/Rversion.h" | sed 's:[^0-9.]*::g'`
 RVER1=`sed -n 's:.*R_MINOR::p' < "$DST/R-fw/R.framework/Headers/Rversion.h" | sed 's:[^0-9.]*::g'`
@@ -99,11 +102,11 @@
 fi
 
 echo " - Building R framework package for $VERFULL ..."
-echo pkgbuild --sign 'Developer ID Installer' --identifier org.R-project.R.fw.pkg --install-location /Library/Frameworks --version "$VER" --scripts "$PKGROOT/scripts-R-fw" --timestamp --root "$DST/R-fw" --component-plist "$PKGROOT/R-fw.plist" "$PKGROOT/R-fw.pkg"
-pkgbuild --sign 'Developer ID Installer' --identifier org.R-project.R.fw.pkg --install-location /Library/Frameworks --version "$VER" --scripts "$PKGROOT/scripts-R-fw" --timestamp --root "$DST/R-fw" --component-plist "$PKGROOT/R-fw.plist" "$PKGROOT/R-fw.pkg"
+echo pkgbuild --identifier org.R-project.R.fw.pkg --install-location /Library/Frameworks --version "$VER" --scripts "$PKGROOT/scripts-R-fw" --timestamp --root "$DST/R-fw" --component-plist "$PKGROOT/R-fw.plist" "$PKGROOT/R-fw.pkg"
+pkgbuild --identifier org.R-project.R.fw.pkg --install-location /Library/Frameworks --version "$VER" --scripts "$PKGROOT/scripts-R-fw" --timestamp --root "$DST/R-fw" --component-plist "$PKGROOT/R-fw.plist" "$PKGROOT/R-fw.pkg"
 
 echo " - Creating distribution description ..."
 "$PKGROOT/mkres" "$PKGROOT"
 
 echo " - Building final bundle ..."
-productbuild --timestamp --resources "$PKGROOT/res" --sign 'Developer ID Installer' --distribution "$PKGROOT/dist.plist" --package-path "$PKGROOT" "$PKGROOT/$NAME.pkg" && cp -p "$PKGROOT/$NAME.pkg" "$BASE/deploy/$osname/$NAME/" && echo "R $VERFULL, GUI $GUIVER in $NAME.pkg" > "$BASE/deploy/$osname/$NAME/$NAME.pkg.SUCCESS" && echo "SUCCESS, a copy is in $BASE/deploy/$osname/$NAME/$NAME.pkg"
+productbuild --timestamp --resources "$PKGROOT/res" --distribution "$PKGROOT/dist.plist" --package-path "$PKGROOT" "$PKGROOT/$NAME.pkg" && cp -p "$PKGROOT/$NAME.pkg" "$BASE/deploy/$osname/$NAME/" && echo "R $VERFULL, GUI $GUIVER in $NAME.pkg" > "$BASE/deploy/$osname/$NAME/$NAME.pkg.SUCCESS" && echo "SUCCESS, a copy is in $BASE/deploy/$osname/$NAME/$NAME.pkg"

Index: R4/src/consh.c
===================================================================
--- R4/src/consh.c	(revision 4304)
+++ R4/src/consh.c	(working copy)
@@ -4,6 +4,7 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 FILE *lf;
 
Index: R4/src/fixup.c
===================================================================
--- R4/src/fixup.c	(revision 4304)
+++ R4/src/fixup.c	(working copy)
@@ -1,4 +1,5 @@
 #include <unistd.h>
+#include <stdlib.h>
 
 int main(int argc, char **argv) {
   setuid(0);
